src/dos/drive_local.cpp

```diff
@@ -121,6 +143,15 @@ bool localDrive::FileOpen(DOS_File **file, char *name, uint32_t flags)
 		DOS_SetError(DOSERR_ACCESS_CODE_INVALID);
 		return false;
 	}
+
+	// Don't allow opening read-only files in write mode.
+	uint16_t test_attr = 0;
+	if (((flags & 0xf) == OPEN_WRITE || (flags & 0xf) == OPEN_READWRITE) &&
+	    (GetFileAttr(name, &test_attr) && (test_attr & DOS_ATTR_READ_ONLY))) {
```

Comment:
Sometimes low-level bit-twiddling is necessary, but whenever we can, we should use the much nicer `bit_view` unions that lead to more readable code, and thus hopefully less obscure bugs. This is a prime cadidate to be restructured to use the `FatAttributeFlags` union instead.

Reply:
At the very least extract `flags & 0xf` into a temp var with a descriptive name. I have no idea what `0xf` does just by looking at this.